imports/plugins/core/graphql/lib/hocs/withTags.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { PropTypes } from "prop-types";
+import { Query } from "react-apollo";
+import getTags from "../queries/getTags";
+import { getTagIds } from "/lib/selectors/tags";
+
+export default (Component) => (
+  class Tags extends React.Component {
+    static propTypes = {
+      shouldSkipGraphql: PropTypes.bool, // Whether to skip this HOC's GraphQL query & data
+    };
+
+    render() {
+      const { shouldSkipGraphql, shopId } = this.props;
+
+      if (shouldSkipGraphql || !shopId) {
+        return (
+          <Component {...this.props} />
+        );
+      }
+
+      const variables = {
+        shopId,
+        isTopLevel: true
+      };
+
+      return (
+        <Query query={getTags} variables={variables}>
+          {({ loading, error, data }) => {
+            const props = {
+              ...this.props,
+              isLoadingTags: loading
+            };
+
+            if (loading === false && data) {
+              let { tags: { nodes: tags} } = data;
+              // tags = _.cloneDeep(tags);
+              _.sortBy(tags, this.props.sortBy || "position"); // puts tags without position at end of array
+              const tagsByKey = {};
```

Comment:
@Akarshit Wondering if it makes sense to pass `tagsByKey` here since you're already passing the array of tags. Would every component using this HOC use `tagsByKey`, or is it specific to `TagNav`?

Reply:
@dancastellon If I don't do that, I will have to maintain tagsByKey in the child component's state or just as a class property(while it is actually a props thing), TBH I don't know which is a better approach.
What do you think should I do?

Reply:
@Akarshit I looked into this a little more. I see how a few components like `tagGroup` and `tagGroupBody` take `tagIds` and `tagsByKey` props. Those components could probably be reworked to just accept `tags[]` since with ES6 you an now do `tag = tags.find(tag => tag._id === tagId)`. Take a look and see if it's relatively straightforward to change. If it's a pain, then maybe leave `tagsByKey` in this HOC. Otherwise, would be nice to clean the components up a little. Up to you :)